src/mapping_navigation/scripts/navigation.py

```diff
@@ -46,20 +65,56 @@ def handle_gps_data(self, position: PoseStamped):
     def handle_speed_data(self, speed: Float64):
         self.car_state.update_speed(speed.data)
 
+    # Checks ahead up to 5 meters scanning for a new road segment.
+    def check_ahead(self, closest_point) -> RoadWarning:
+        for point in self.path[closest_point + 1:]:
+            if point.segment_type != self.path[closest_point].segment_type and point.segment_type is not None:
+                if self.car_state.calc_distance(point.coordinate[0], point.coordinate[1]) < 5:
+                    rospy.loginfo(
+                        f'The following road segment is 5 meters or less ahead {RoadWarning(point.segment_type.value)}')
+                    return RoadWarning(point.segment_type.value)
+            if self.car_state.calc_distance(point.coordinate[0], point.coordinate[1]) > 5:
+                break
+
+        return RoadWarning.SAME_AHEAD
+
+    # Get the closet point to the car
+    # Pure pursuit does not always keep track of the closest point so this is required to find it
+    def find_nearest_point(self) -> int:
+        dx = [self.car_state.rear_x - point.coordinate[0] for point in self.path]
+        dy = [self.car_state.rear_y - point.coordinate[1] for point in self.path]
+        distances = np.hypot(dx, dy)
+        index = int(np.argmin(distances))
+        return index
+
     """
     Returns a steering angle for the car in Airsim range of -1 to 1
     """
 
     def get_steering_angle(self) -> float:
-        ackerman_angle_rad, self.target_index = self.navigator.pure_pursuit_steer_control(self.car_state)
+        ackerman_angle_rad, target_index = self.navigator.pure_pursuit_steer_control(self.car_state)
+        if self.target_index != target_index:
+            self.last_index = self.target_index
+            self.target_index = target_index
         ackerman_angle_ratio = math.degrees(ackerman_angle_rad) / 45
-        rospy.loginfo(f"""Target point is {path[self.target_index].coordinate[0]}, {path[self.target_index].coordinate[1]},
-                      car position {self.car_state.point.coordinate[0]}, {self.car_state.point.coordinate[1]}""")
+        rospy.loginfo(
+            f"Target point is {self.path[self.target_index].coordinate[0]}, {self.path[self.target_index].coordinate[1]}")
+        rospy.loginfo(f"car position {self.car_state.point.coordinate[0]}, {self.car_state.point.coordinate[1]}")
         rospy.loginfo(f'Current steering angle in degrees {math.degrees(ackerman_angle_rad)}')
         return ackerman_angle_ratio
 
 
+# Can use either of the path methods by uncommenting one or the other if you want to test with the old path
 if __name__ == "__main__":
-    path = read_points(os.path.abspath(os.path.dirname(__file__)) + "/coords.txt")
-    navigation = Navigation(look_ahead_distance=4.0, look_forward_gain=0.1, path=path, wheel_base=2.2)
+
+    # map = load_from_file(os.path.abspath(os.path.dirname(__file__)) + "/new_coords.txt")
```

Comment:
I left two paths, the new one with the new path data, and the old one with no path data. Currently I just left the new method commented out so it would not mess with other default testing. But if you want to use my new path, which is a little wonky, or define your own new path uncomment the new method and comment out the old method.

In another pr this should be controlled by and environment var

Reply:
Is the new path the same as the old one? Curious why it's a binary file

Reply:
Nope completely new, its created by the mapping tool, those maps are stored with pythons serializer

Reply:
I would recommend creating your own new test path, the one I made is a little jittery.